onrobot_rg2_driver/src/rg2_driver_node.py

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/python3
+from modbus_tk.modbus import ModbusError
+from threading import Lock
+
+import rospy
+
+from sensor_msgs.msg import JointState
+
+from onrobot_rg2_driver.srv import GripperState, GripperStateResponse, GripperStateRequest
+
+from rg2_modbus import Rg2Modbus
+
+
+class Rg2DriverNode:
+    MIN_RG2_FORCE: float = 3.0  # [N]
+    MAX_RG2_FORCE: float = 40.0  # [N]
+    MIN_RG2_RANGE: float = 0.0  # [N]
+    MAX_RG2_RANGE: float = 100.0  # [N]
+    TIMEOUT: float = 10.0  # [s]
+
+    def __init__(self) -> None:
+        rospy.init_node("rg2_driver_node", anonymous=False)
+
+        ur_robot_ip = rospy.get_param('~ur_robot_ip', '10.42.0.49')
+        frame_id = rospy.get_param('~base_link_frame', 'rg2_base_link')
+        joint_name = rospy.get_param('~gripper_joint_name', 'rg2_gripper_joint')
+        self.service_call_timeout_sec = rospy.get_param('~service_call_timeout_sec', 10.0)
+        self._rg2_force_range = (3.0, 40.0)  # [N]
+        self._lock = Lock()
+
+        self._rg2_modbus = Rg2Modbus(ur_robot_ip)
+
+        self._joint_state_msg = JointState()
+        self._joint_state_msg.header.frame_id = frame_id
+        self._joint_state_msg.name = [joint_name]
+
+        rate = rospy.Rate(0.5)
+        while not self._rg2_modbus.watchdog.is_active():
+            rospy.loginfo_throttle(5.0, 'Waiting for URscript Modbus watchdog activation...')
+            rate.sleep()
+
+        self._joint_publisher = rospy.Publisher("/joint_states", JointState, queue_size=1)
+
+        self._set_gripp_state_service = rospy.Service(
+            "rg2/set_gripper_width", GripperState, self._set_grip_state
+        )
+
+        rospy.Timer(rospy.Duration(0.05), self._pub_joint_state_cb)
+
+        rospy.loginfo(f'[{rospy.get_name()}] Node started')
+
+    def _set_grip_state(self, req: GripperStateRequest) -> GripperStateResponse:
+        try:
+            with self._lock:
```

Comment:
```suggestion
        with self._lock:
            try:        
```
I'd rather do it this way